### ISSUE TITLE
[Bug]: wallet_addEthereumChain() does not support HTTP RPCs

### ISSUE THREAD

**Issue by 0xfoobar** [MetaMask/metamask-extension#14416]
### Describe the bug

When you run

```
await ethereum.request({
        method: 'wallet_addEthereumChain',
        params: [
          {
            chainId: '0x539',
            chainName: 'MyTestnet',
            rpcUrls: ['http://1.1.1.1:8545'],
          },
        ],
      });
```

it errors out with `Expected an array with at least one valid string HTTPS url 'rpcUrls', Received: http://1.1.1.1:8545`.

However, the user is able to manually add an HTTP url within their own RPC settings, even though it fails on programmatic adding. For consistency, Metamask should allow programmatically adding an HTTP url.

Additionally, [EIP-3085]([](https://eips.ethereum.org/EIPS/eip-3085#returns)All URL strings MUST include the protocol component of the URL. HTTPS SHOULD always be used over HTTP.) specifies HTTPS urls as a SHOULD, not a MUST. Given the prevalence of running local clients and exposing ports without SSL certificates, this would be a great improvement for users interested in self-hosting.

The change would need to be made here: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js#L86

The fix is simple: check for an HTTP or HTTPS uri, instead of just an HTTPS uri.

### Steps to reproduce

Run within the JS console:

```
await ethereum.request({
        method: 'wallet_addEthereumChain',
        params: [
          {
            chainId: '0x539',
            chainName: 'MyTestnet',
            rpcUrls: ['http://1.1.1.1:8545'],
          },
        ],
      });
```

### Version

10.12.3

### Browser

Chrome

### Operating system

MacOS

**Comment by 0xfoobar** [comment]
Anybody home?

**Comment by wizzard0** [comment]
It makes sense to allow at least http://127.0.0.1 here

**Comment by 0xfoobar** [comment]
Or all HTTP RPCs, because the spec allows them.

**Comment by wizzard0** [comment]
> Or all HTTP RPCs, because the spec allows them.
Yeah. I'd argue it makes sense to show a prominent warning for http://some-internet-ip though.

**Comment by P233** [comment]
+1

**Comment by BelfordZ** [comment]
localhost will work now as of https://github.com/MetaMask/metamask-extension/pull/14272 

Making it work for localhost is a no-brainer - its not reasonable to put ssl on a locally running node.

That said, for a remotely hosted node, you absolutely should be using ssl, and as of right now there are no plans to loosen this requirement. 

Hope that helps.
